Fix user upload bug

NoahCarnahan opened the pull request:
http://manage.dimagi.com/default.asp?203170

User uploads are failing when name cells look like numerical data, because openpyxl returns the values as numeric types. Fixed by calling `unicode()` on the values.
@snopoke

NoahCarnahan committed:
Make sure name is text

NoahCarnahan commented:
Hold off on merge, going to add a test

NoahCarnahan committed:
Unbreak running individual tests in TestUserBulkUpload

NoahCarnahan committed:
Add tests for numeric user name fix

snopoke commented:
:boot: :construction:

czue commented:
Looks real:
```
======================================================================
FAIL: test_location_update (corehq.apps.users.tests.bulk_upload.TestUserBulkUpload)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/users/tests/bulk_upload.py", line 142, in test_location_update
    self.assertEqual(self.user.location_id, self.location._id)
AssertionError: None != u'7eca110c0d0228fd39c2e3b91b511ee9'
```

NoahCarnahan committed:
Don't reuse user between tests

Because some of the tests change the user's properties

NoahCarnahan committed:
Don't memoize getting the user

It results in stale properties for self.user

NoahCarnahan committed:
Fix stale privileges bug

Because domain_has_privilege() is quickcached, it will return
stale values. The patch ensures it always returns true. There
is also no longer a need to update the domain's privileges,
since we are returning true for all of them now.

NoahCarnahan committed:
cleanup imports

NoahCarnahan commented:
Lots of state was being preserved between test cases. These new commits break that coupling.

czue commented:
nice
